nhl_api_py/core/models.py

```diff
@@ -125,6 +125,84 @@ def from_dict(cls, data: dict):
         return cls(**final_data)
 
 
+@dataclass
+class GeneralGame(Model):
+    """
+    Represents and contains all data for a given game, returned from the NHL API.
+    """
+
+    pk: Optional[int] = None
+    type: Optional[str] = None
+    date_time: Optional[str] = None
+    end_date_time: Optional[str] = None
+    abstract_game_state: Optional[str] = None
+    coded_game_state: Optional[str] = None
+    detailed_state: Optional[str] = None
+    status_code: Optional[str] = None
+    start_time_tbd: Optional[bool] = None
+    away: Optional[Team] = None
+    home: Optional[Team] = None
+    plays: Optional[dict] = None  # Needs to be converted to list(Play)
+
+    @classmethod
+    def from_dict(cls, data: dict):
+        converted_data = convert_keys_to_snake_case(data)
+
+        # Extract nested data after top-level if it exists
+        game_data = converted_data.get("game_data", dict())
+        live_data = converted_data.get("live_data", dict())
+
+        game = _field_only_keys(game_data.get("game", dict()), cls)
+        datetime_data = _field_only_keys(game_data.get("datetime", dict()), cls)
+        status_data = _field_only_keys(game_data.get("status", dict()), cls)
+
+        # Extract one nested dict further for teams data if it exists
+        teams_data = _field_only_keys(game_data.get("teams", dict()), cls)
+        away_data = _field_only_keys(teams_data.get("away", dict()), Team)
+        away_data = Team.from_dict(away_data) if len(away_data) != 0 else None
+        home_data = _field_only_keys(teams_data.get("home", dict()), Team)
+        home_data = Team.from_dict(home_data) if len(home_data) != 0 else None
+
+        play_data = _field_only_keys(live_data.get("plays", dict()), cls)
+        play_data = _field_only_keys(play_data.get("all_plays", dict()), Play)
+        play_data = Play.from_dict(play_data) if len(play_data) != 0 else None
+        final_data = {
+            **game,
+            **datetime_data,
+            **status_data,
+            "away": away_data,
+            "home": home_data,
+            "plays": play_data,
+        }
+        return cls(**final_data)
+
+
+@dataclass
+class BoxscoreGame(Model):
```

Comment:
Lets just call this `Boxscore`
```suggestion
class Boxscore(Model):
```